ejb/WFLY-6892-ejb-access-logging.adoc

```diff
@@ -0,0 +1,138 @@
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* implement the access logging feature similar to WildFly web access logging.
+
+* add access-logging definition to wildfly-ejb3 schema (see next section below)
+
+* configure ejb access logging with CLI
+
+* enable and disable ejb access logging
+
+* configure how to rotate ejb access logs if applicable
+
+** default should be OFF for performance consideration
+
+* configure output location and format
+
+** directory/file
+
+** stdout
+
+* choose from pre-defined logging formats (need to adapt for ejb invocation):
+
+** common (default): %h %l %u %t "%r" %s %b
+
+** combined: %h %l %u %t "%r" %s %b "%{i,Referer}" "%{i,User-Agent}"
+
+* for more customization, define which attributes to include in ejb access logging
+
+** consider adapting the following attributes that are currently supported by web access logging:
+
+    %a - Remote IP address
+    %A - Local IP address
+    %b - Bytes sent, excluding HTTP headers, or '-' if no bytes were sent
+    %B - Bytes sent, excluding HTTP headers
+    %h - Remote host name
+    %H - Request protocol
+    %l - Remote logical username from identd (always returns '-')
+    %m - Request method
+    %p - Local port
+    %q - Query string (excluding the '?' character)
+    %r - First line of the request
+    %s - HTTP status code of the response
+    %t - Date and time, in Common Log Format format
+    %u - Remote user that was authenticated
+    %U - Requested URL path (ejb invocation equivalent)
+    %v - Local server name
+    %D - Time taken to process the request, in millis
+    %T - Time taken to process the request, in seconds
+    %I - current Request thread name (can compare later with stacktraces)
+
+=== New elements in wildfly-ejb3_6_0.xsd schema
+ejb30 subsystem is extended with a new top-level element access-log, defined as follows:
+[source,xml]
+----
+    <xs:complexType name="accessLogType">
+        <xs:attribute name="pattern" use="optional" type="xs:string" default="common"/>
+        <xs:attribute name="worker" use="optional" type="xs:string" default="default"/>
+        <xs:attribute name="directory" use="optional" type="xs:string" default="${jboss.server.log.dir}"/>
+        <xs:attribute name="relative-to" use="optional" type="xs:string" />
+        <xs:attribute name="prefix" use="optional" type="xs:string" default="access_log."/>
+        <xs:attribute name="suffix" use="optional" type="xs:string" default="log"/>
+        <xs:attribute name="rotate" use="optional" type="xs:string" default="true"/>
+        <xs:attribute name="use-server-log" use="optional" type="xs:string" default="false"/>
+        <xs:attribute name="extended" use="optional" type="xs:string" default="false" />
+        <xs:attribute name="predicate" use="optional" type="xs:string" />
+    </xs:complexType>
+----
+
+=== Nice-to-Have Requirements
+
+* users should be able to configure condition/filter to determine which requests to log
```

Comment:
If you end up supporting this I'd like to see some detailed requirements on how this would be done.

Reply:
I was thinking using the predicate attribute for some basic filtering, either by reusing the undertow predicate impl, or writing one more tailored to ejb invocations.

Reply:
That seems fine. I only mentioned this because if the output is used by a log aggregator I think it most cases it makes sense for the aggregator to make those determinations. However there could be performance penalties if we're logging heaps of data.